tasks/queue-granules/index.js

```diff
@@ -35,9 +35,11 @@ async function fetchGranuleProvider(prefix, providerId) {
  */
 function groupAndBatchGranules(granules, batchSize) {
   const filteredBatchSize = isNumber(batchSize) ? batchSize : 1;
+
   const granulesByCollectionMap = groupBy(
     granules,
-    (g) => constructCollectionId(g.dataType, g.version)
+    (g) => (g.collectionId !== undefined ? g.collectionId
+      : (constructCollectionId(g.dataType, g.version)))
```

Comment:
What do you think of doing this?

```
function getDataTypeAndVersionFromGranule(granule) {
  if (granule.dataType && granule.version) {
    const { dataType, version } = granule;
    return { dataType, version };
  }
  if (collectionId) {
    const [dataType, version] = deconstructCollectionId(collectionId);
    return { dataType, version }
  }
  throw new Error('Invalid collection, please check task input to make sure collection information is provided on each granule');
}
```

Then here you can use it and below you can use it again:

```suggestion
    (g) => { 
      const { dataType, version } = getDataTypeAndVersionFromGranule(g);
      return constructCollectionId(dataType, version);
    },
```